server/src/send-update/send-update.controller.ts

```diff
@@ -0,0 +1,79 @@
+import { Controller, Param, Post, Req, Res } from "@nestjs/common";
+import { ConfigService } from "../config/config.service";
+import { SendUpdateService } from "./send-update.service";
+import { ProjectService } from "../project/project.service";
+import { ListservAuthService } from "../listserv/auth.service";
+import { Request } from "express";
+
+@Controller()
+export class SendUpdateController {
+  apiKey = "";
+  list = "";
+  sendgridEnvironment = "";
+
+  constructor(
+    private readonly config: ConfigService,
+    private readonly sendUpdateService: SendUpdateService,
+    private readonly projectService: ProjectService,
+    private readonly listservAuthService: ListservAuthService
+  ) {
+    this.apiKey = this.config.get("SENDGRID_API_KEY");
+    this.list = this.config.get("SENDGRID_LIST");
+    this.sendgridEnvironment = this.config.get("SENDGRID_ENVIRONMENT")
+  }
+
+  @Post("/projects/:id/send-update")
+  async sendUpdate(@Param() params, @Req() request: Request, @Res() response) {
+    const validatedUser = await this.listservAuthService.validateUser(request.headers.authorization)
```

Comment:
Nest recommends handling things like authentication with [Guards](https://docs.nestjs.com/guards). There's an example of that in [applicant portal](https://github.com/NYCPlanning/labs-applicant-portal/blob/main/server/src/authenticate.guard.ts) but you might find Nest's docs [here](https://docs.nestjs.com/security/authentication#implementing-the-authentication-guard) and [here](https://docs.nestjs.com/recipes/passport#extending-guards) more helpful. Note those examples use Passport but they should still be helpful even if you aren't using it. Those examples still have `AuthService` so the guard you add will likely use your service, not replace it. Once you have a Guard, you can add it to this route with a decorator instead of validated the call in the controller itself.

Reply:
I agree the guard is unlikely to get used more than once but I still think it's worth it - there's value in doing our best to stick to the framework's best practices and given the guard should be relatively simple to implement I think it's worth it. Plus, it'll be good experience for you with a framework we're planning to use more in the future. If you find yourself spending more than an afternoon getting the same functionality to work with a Guard, then we'll talk.